components/core/tests/test-ffi_SchemaTree.cpp

```diff
@@ -19,41 +17,63 @@ namespace {
  */
 [[nodiscard]] auto insert_node(
         SchemaTree& schema_tree,
-        SchemaTree::NodeLocator locator,
-        SchemaTreeNode::id_t expected_id
+        SchemaTree::NodeLocator const& locator,
+        SchemaTree::Node::id_t expected_id
 ) -> bool;
 
 /**
  * @param schema_tree
  * @param locator
  * @param expected_id
- * @return Whether the node exists and its ID matches the expected ID.
+ * @return Whether the node exists with its ID matches the expected ID and its information matches
+ * the ones specified in the locator.
  */
 [[nodiscard]] auto check_node(
         SchemaTree const& schema_tree,
-        SchemaTree::NodeLocator locator,
-        SchemaTreeNode::id_t expected_id
+        SchemaTree::NodeLocator const& locator,
+        SchemaTree::Node::id_t expected_id
 ) -> bool;
 
 auto insert_node(
         SchemaTree& schema_tree,
-        SchemaTree::NodeLocator locator,
-        SchemaTreeNode::id_t expected_id
+        SchemaTree::NodeLocator const& locator,
+        SchemaTree::Node::id_t expected_id
 ) -> bool {
     return false == schema_tree.has_node(locator)
            && expected_id == schema_tree.insert_node(locator);
 }
 
 auto check_node(
         SchemaTree const& schema_tree,
-        SchemaTree::NodeLocator locator,
-        SchemaTreeNode::id_t expected_id
+        SchemaTree::NodeLocator const& locator,
+        SchemaTree::Node::id_t expected_id
 ) -> bool {
     auto const node_id{schema_tree.try_get_node_id(locator)};
-    return node_id.has_value() && node_id.value() == expected_id;
+    if (false == node_id.has_value() || node_id.value() != expected_id) {
+        // Check node ID
+        return false;
+    }
+    auto const& node{schema_tree.get_node(expected_id)};
+    if (node.is_root()) {
+        // The node after initialization must not be the root.
+        return false;
+    }
+    auto const optional_parent_id{node.get_parent_id()};
+    if (false == optional_parent_id.has_value()) {
+        // Non-root node must have a parent ID.
+        return false;
+    }
+    if (optional_parent_id.value() != locator.get_parent_id()
+        || node.get_type() != locator.get_type() || node.get_key_name() != locator.get_key_name())
```

Comment:
Technically this is implicitly checked by `try_get_node`.

Reply:
that is correct, but adding this in the unit test is also checking the functionality of `try_get_node` works properly right?